src/types.js

```diff
@@ -0,0 +1,17 @@
+import PropTypes from 'prop-types';
+
+export const userType = PropTypes.shape({
+  name: PropTypes.string.isRequired,
+  email: PropTypes.string.isRequired,
+  address: PropTypes.shape({
+    street: PropTypes.string.isRequired,
+  }).isRequired,
+});
+
+export const commentType = PropTypes.arrayOf(
+  PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    body: PropTypes.string.isRequired,
+    email: PropTypes.string.isRequired,
+  }),
+);
```

Comment:
```suggestion
export const commentType = PropTypes.shape({
    name: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  });
```

Reply:
comment is object

Reply:
comment its array with objects)))

Reply:
okay, I see, it's better to define a single instance
you know you have an array of objects, so you define a type for a single object and then in component add it to PropTypes.arrayOf(thisObject)
I believe, it's a more universal approach🙃